src/commands/trace/trace.ts

```diff
@@ -0,0 +1,144 @@
+    envVarTags: process.env.DD_TAGS,
+  }
+
+  private name?: string
+
+  private noFail?: boolean
+
+  public async execute() {
+    if (!this.command || !this.command.length) {
+      this.context.stderr.write('Missing command to run\n')
+
+      return 1
+    }
+
+    const [command, ...args] = this.command
+    const id = crypto.randomBytes(5).toString('hex')
+    const startTime = new Date().toISOString()
+    const spawnResult = spawnSync(command, args, {env: {...process.env, DD_CUSTOM_PARENT_ID: id}, stdio: 'inherit'})
+    const endTime = new Date().toISOString()
+    const exitCode = spawnResult.status ?? this.signalToNumber(spawnResult.signal) ?? badCommandExitCode
+    const [ciEnvVars, provider] = this.getCIEnvVars()
+    if (provider) {
+      const api = this.getApiHelper()
+      const commandStr = this.command.join(' ')
+      await api.reportCustomSpan(
+        {
+          command: commandStr,
+          custom: {
+            id,
+            parent_id: process.env.DD_CUSTOM_PARENT_ID,
+          },
+          data: ciEnvVars,
+          end_time: endTime,
+          is_error: exitCode !== 0,
+          name: this.name ?? commandStr,
+          start_time: startTime,
+          tags: this.config.envVarTags ? parseTags(this.config.envVarTags.split(',')) : {},
+        },
+        provider
+      )
+    }
+
+    return exitCode
+  }
+
+  public getCIEnvVars(): [Record<string, string>, Provider?] {
+    if (process.env.CIRCLECI) {
+      return [
+        this.getEnvironmentVars([
+          'CIRCLE_BRANCH',
+          'CIRCLE_BUILD_NUM',
+          'CIRCLE_BUILD_URL',
+          'CIRCLE_JOB',
+          'CIRCLE_NODE_INDEX',
+          'CIRCLE_NODE_TOTAL',
+          'CIRCLE_PROJECT_REPONAME',
+          'CIRCLE_PULL_REQUEST',
+          'CIRCLE_REPOSITORY_URL',
+          'CIRCLE_SHA1',
+          'CIRCLE_TAG',
+          'CIRCLE_WORKFLOW_ID',
+        ]),
+        CIRCLECI,
+      ]
+    }
+    const errorMsg = `Cannot detect any supported CI Provider. This command only works if run as part of your CI. Supported providers: ${SUPPORTED_PROVIDERS}.`
+    if (this.noFail) {
+      this.context.stdout.write(
+        `${chalk.yellow.bold('[WARNING]')} ${errorMsg} Not failing since the --no-fail options was used.\n`
+      )
+
+      return [{}]
+    } else {
+      throw new Error(errorMsg)
+    }
+  }
+
+  private getApiHelper(): APIHelper {
+    if (!this.config.apiKey) {
+      this.context.stdout.write(
+        `Neither ${chalk.red.bold('DATADOG_API_KEY')} nor ${chalk.red.bold('DD_API_KEY')} is in your environment.\n`
+      )
+      throw new Error('API key is missing')
+    }
+
+    return apiConstructor(this.getBaseIntakeUrl(), this.config.apiKey)
+  }
+
+  private getBaseIntakeUrl() {
+    if (process.env.DATADOG_SITE || process.env.DD_SITE) {
+      return `https://webhooks-http-intake.logs.${process.env.DATADOG_SITE || process.env.DD_SITE}`
+    }
+
+    return 'https://webhooks-http-intake.logs.datadoghq.com'
```

Comment:
I believe this was going to check. Can you double check with @AdrianLC ?

Reply:
```suggestion
    return 'https://webhook-http-intake.datadoghq.com'
```